pkg/kp/pcstore/consul_store_test.go

```diff
@@ -244,10 +244,54 @@ func TestWatch(t *testing.T) {
 	})
 
 	quit := make(chan struct{})
-	watch := store.Watch(pod, quit)
+	watch := store.Watch(quit)
 
 	var watched *WatchedPodCluster
-	pc, err := store.Create(podID, az, clusterName, selector, annotations)
+	session := kptest.NewSession()
+	pc, err := store.Create(podID, az, clusterName, selector, annotations, session)
+	if err != nil {
+		t.Fatalf("Unable to create pod cluster: %s", err)
+	}
+
+	select {
+	case watchedPC := <-watch:
+		watched = &watchedPC
+	case <-time.After(5 * time.Second):
+		quit <- struct{}{}
+		t.Fatal("nothing on the channel")
+	}
+
+	if watched == nil {
+		t.Fatalf("Expected to get a watched PodCluster, but did not")
+	}
+
+	if watched.PodCluster.ID != pc.ID {
+		t.Fatalf("Expected watched PodCluster to match %s Pod Cluster ID. Instead was %s", pc.ID, watched.PodCluster.ID)
+	}
+}
+
+func TestWatchPodCluster(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	pod := fields.ID("pod_id")
+	podID := types.PodID("pod_id")
+	az := fields.AvailabilityZone("us-west")
+	clusterName := fields.ClusterName("cluster_name")
+
+	selector := klabels.Everything().
+		Add(fields.PodIDLabel, klabels.EqualsOperator, []string{pod.String()}).
+		Add(fields.AvailabilityZoneLabel, klabels.EqualsOperator, []string{az.String()}).
+		Add(fields.ClusterNameLabel, klabels.EqualsOperator, []string{clusterName.String()})
+
+	annotations := fields.Annotations(map[string]interface{}{
+		"foo": "bar",
+	})
+
+	quit := make(chan struct{})
+	watch := store.WatchPodCluster(pod, quit)
```

Comment:
actually now I am slightly confused. I would have thought that instead of `pod`, I would need to pass in `pc.ID` here (where `pc` is the result form the `Create` a few lines below)? May need someone to help shed some light on how this works.